streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNodePunctuator.java

```diff
@@ -19,6 +19,7 @@
 import org.apache.kafka.streams.processor.PunctuationType;
 import org.apache.kafka.streams.processor.Punctuator;
 
+@FunctionalInterface
```

Comment:
This is an internal non-user facing interface -- so seems we don't gain much by adding the annotation?

Reply:
NP. In Kafka Streams code base, we have a strict separation between public and non-public, and everything in packages called `internals` is non-public.